### PR TITLE
Add .js extension support with es6 example

### LINKED ISSUE #214: Support recognizing TypeScript files imported with `.js` suffix
In TypeScript, it is valid to import TypeScript files with `.js` extension. TypeScript will match the coresponding `.ts` files if they exist. This behaviour is commonly used in Node.js projects as [`.js` file extensions are required in Node.js ESM imports](https://nodejs.org/api/esm.html#esm_mandatory_file_extensions).

It would be great if ts-unused-exports also recognize variable imported with `.js` suffix.

Related: https://github.com/evanw/esbuild/issues/1343#issuecomment-854055814, https://github.com/microsoft/TypeScript/issues/16577#issuecomment-703190339, https://github.com/microsoft/TypeScript/issues/16577#issuecomment-754941937

## Example

In [this example project derived from `example/simple`](https://github.com/mon-jai/ts-unused-exports/tree/master/example/simple), `inc()` from `math.ts` was imported with `.js` extension.

https://github.com/pzavolinsky/ts-unused-exports/blob/ad208c946855f9f5456d3d20435a4282f946c72b/example/simple/app.ts#L1

And TypeScript recognize it sucessfully.

However, ts-unused-exports failed to recognize that the default import of`math.ts` was imported in `app.ts`.

Screenshot 2022-02-19 000527

### PR DESCRIPTION
**Pull request by mrseanryan**
Fixes #214

### CODE REVIEW

File: src/parser/util.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ export const indexCandidates = [
 ];
 
 export function removeExportStarPrefix(path: string): string {
```
[the remark this answers is not in the document]
Comment (mrseanryan) [reply src/parser/util.ts]: removeExportStarPrefix() looks OK as-is

File: example/simple/execute.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+npx tsc && node ./lib/app.js
```
[the remark this answers is not in the document]
Comment (mrseanryan) [reply example/simple/execute.sh:1]: - could add unit tests for the various extensions (at least for index...)

**Comment by mon-jai** [comment]
I don't know if TypeScript allow .js extension if target or module is not es*.

In case TypeScript not allowing it, we may need to add a check and mirror TypeScript's semantic.

**Comment by mrseanryan** [comment]
> I don't know if TypeScript allow .js extension if `target` or `module` is not `es*`.
> 
> In case TypeScript not allowing it, it may be better to add a check and mirror TypeScript's semantic.

hm - Im thinking it is simplest to *always* allow for js imports - since tsu does not need to build the project, just scan it.

Internally we were already partly doing this by allowing for index.js 

if there is some build error, the dev will fix, then run tsu again…

**Comment by mrseanryan** [comment]
Looks good for major release - let me know if you have more feedback or ideas !

**Pull request merged**